internal/plugincommon/discovery/processor.go

```diff
@@ -182,28 +180,65 @@ func aggregateObservations(
 	return obs
 }
 
-// Outcome comes to consensus on the contract addresses and updates the chainreader. It doesn't actually
-// return an Outcome.
+// getRouterConsensus figures out a single router address from the multiple observed addresses from multiple source
+// chains. This happens in two phases, first we get the consensus on the router address for each source chain
+// according the threshold (fChain). Second, once we have consensus across the source chains, we select whichever
+// address was observed the most times. If there is a tie, we return an error.
+//
+// For a typical configuration, we expect all source chains to report the same destination router address.
+func getRouterConsensus(
+	lggr logger.Logger,
+	routerAddrs map[cciptypes.ChainSelector][][]byte,
+	thresh consensus.MultiThreshold[cciptypes.ChainSelector],
```

Comment:
That wont work, suppose the dest chain has F=5 and for one of the sources has F=1 because there are only 3 readers. That source chain would never reach consensus. You could take it one step further and imagine all source readers have F=1, in this case there would never be consensus from any source readers if F=5

Reply:
Not sure I understand, why would there never be consensus? This is a purely dest chain read and no sources are touched.

Reply:
The dest router address is being discovered by reading from the source chain onramp's